include/arc_utilities/arc_helpers.hpp

```diff
@@ -586,6 +588,63 @@ namespace arc_helpers
         }
     }
 
+    template<typename ColorType>
+    inline ColorType Multiply(const ColorType& color, const float factor)
+    {
+        ColorType ret;
+        ret.r = TrimColorValue(factor * color.r);
+        ret.g = TrimColorValue(factor * color.g);
+        ret.b = TrimColorValue(factor * color.b);
+        ret.a = TrimColorValue(factor * color.a);
+        return ret;
+    }
+
+    template<typename ColorType>
+    inline ColorType Multiply(const float factor, const ColorType& color)
+    {
+        return color * factor;
+    }
+
+    template<typename ColorType>
+    inline ColorType Add(const ColorType& c1, const ColorType& c2)
+    {
+        ColorType ret;
+        ret.r = TrimColorValue(c1.r + c2.r);
+        ret.g = TrimColorValue(c1.g + c2.g);
+        ret.b = TrimColorValue(c1.b + c2.b);
+        ret.a = TrimColorValue(c1.a + c2.a);
+        return ret;
+    }
+
+    /**
+     * @brief InterpolateColor
+     * @param c1 Start color for interpolation
+     * @param c2 End color for interpolation
+     * @param ratio Value between 0 and 1, indicating how far to move from c1 towards c2
+     * @return
+     */
+    template<typename ColorType>
+    inline ColorType InterpolateColor(const ColorType& c1, const ColorType& c2, const float& ratio)
+    {
+        // Safety check ratio
+        float real_ratio = ratio;
+        if (real_ratio < 0.0)
+        {
+            real_ratio = 0.0;
+            std::cerr << "Interpolation ratio < 0.0, set to 0.0" << std::endl;
+        }
+        else if (real_ratio > 1.0)
+        {
+            real_ratio = 1.0;
+            std::cerr << "Interpolation ratio > 1.0, set to 1.0" << std::endl;
+        }
+        // Interpolate
+        // This is the numerically stable version, rather than  (c1 + (c2 - c1) * real_ratio)
+        return Add(Multiply(c1, 1.0 - real_ratio), Multiply(c2, real_ratio));
```

Comment:
Consider doing interpolation in another space (convert to then convert back): http://i.imgur.com/rSB2ji3.gif